source/blog/2017-12-13-compose-custom-ostree.md

```diff
@@ -0,0 +1,185 @@
+---
+title: Composing custom OSTree update for Fedora 27 Atomic Host
+author: sinnykumari
+date: 2017-12-13
+published: true
+comments: false
+tags: atomic, fedora, multi-arch, ostree
+---
+
+With Fedora 27, we ship [Atomic Host on multiple architectures](http://www.projectatomic.io/blog/2017/11/fedora-atomic-27-on-multiarch/) which includes aarch64, ppc64le, and x86_64. We also provide Atomic Host updates for these arches every two weeks so that we can provide a tested, stable, and up-to-date OS. Fedora Atomic Host is built using traditional RPM packages available in the Fedora repository. For example, Fedora 27 Atomic Host is formed using  packages from the Fedora 27 repository which are further composed into [OSTree repository](https://ostree.readthedocs.io/en/latest/manual/introduction/) using rpm-ostree. It is possible to compose and host your own custom-built OSTree repository containing additional features. This article will further guide you on how to compose your own custom OSTree repository and update system from your own hosted OSTree repo.
+
+# Composing and hosting custom Atomic OSTree
+We will be composing a custom Fedora 27 OSTree repository on x86_64. For ease of use, we will compose and host OSTree repo in a Docker container. You can also compose it on a Virtual Machine or bare metal.
+> `Note:` If you are trying this on aarch64 or ppc64le, replace any mention of x86_64 with the right architecture value wherever used in commands.
+
+## Requirement
+- A virtual machine or bare metal system with Fedora installed, preferably Fedora 27 which is the latest stable release.
+- Docker installed and service running on the machine.
+
+Install Docker and run its service on the system where you are planning to run the container:
+
+```
+$ sudo dnf install docker
+$ sudo systemctl start docker
+```
+
+## Dockerfile
+Create a Dockerfile with the following content which we will use to build a Docker container using Fedora27 as its base and on top of it some additional setup is being done.
+
+```
+$ cat Dockerfile
+FROM fedora:27
+
+# Update, install specified packages and clean cached information
+RUN dnf update -y && dnf install -y git python rpm-ostree; \
+dnf clean all
+
+# Create specified directories
+RUN mkdir -p /srv /srv/repo /srv/cache
+
+# Set /srv as working directory and clone fedora-atomic repository into it
+WORKDIR /srv
+RUN git clone https://pagure.io/fedora-atomic.git
+
+# Initialize a new empty repository in /srv/repo/ directory in archive mode
+RUN ostree --repo=repo init --mode=archive-z2
```

Comment:
This can just be `archive` now.

Reply:
Thanks @cgwalters! Yup, I tried replacing --mode=archive-z2 to --mode=archive and it works as well.

@sanjacloud Will it be possible to edit line 46 from "RUN ostree --repo=repo init --mode=archive-z2" to "RUN ostree --repo=repo init --mode=archive"? Since, It's already been merged, I will have to create a new PR.